src/Tribe/Views/V2/Views/Day_View.php

```diff
@@ -37,119 +36,83 @@ class Day_View extends View {
 
-		// Remove any date related arguments.
-		unset( $current_args['date_overlaps'] );
+		$one_day     = new \DateInterval( 'P1D' );
+		$url_date    = Dates::build_date_object( $date )->add( $one_day );
+		$latest      = tribe_get_option( 'latest_date', $url_date );
+		$latest_date = Dates::build_date_object( $latest )->setTime( 0, 0, 0 );
 
-		// Use the updated repository arguments.
-		$current_repository = tribe_events()->by_args( $current_args );
+		if ( $url_date > $latest_date ) {
+			$url = '';
+		} else {
+			$url = $this->build_url_for_date( $url_date, $canonical, $passthru_vars );
+		}
 
-		$default_date   = 'today';
-		$date           = $this->context->get( 'event_date', $default_date );
-		$event_date_var = $default_date === $date ? '' : $date;
+		return $this->filter_next_url( $canonical, $url );
+	}
 
-		$one_day = new \DateInterval( 'P1D' );
-		$next_date = Dates::build_date_object( $date )->add( $one_day )->format( Dates::DBDATEFORMAT );
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function setup_repository_args( \Tribe__Context $context = null ) {
+		$context = null !== $context ? $context : $this->context;
 
-		$next = clone $current_repository;
-		$start = tribe_beginning_of_day( $next_date );
-		$end   = tribe_end_of_day( $next_date );
-		$next->where( 'date_overlaps', $start, $end )->order( 'DESC' )->per_page( 1 );
+		$args = parent::setup_repository_args( $context );
 
-		$next_event = $next->first();
-		$has_next = $next->found();
+		$context_arr = $context->to_array();
 
-		if ( ! $has_next ) {
-			$next = clone $current_repository;
-			$next->where( 'starts_after', tribe_end_of_day( $date ) )->order( 'DESC' )->per_page( 1 );
+		$date = Arr::get( $context_arr, 'event_date', 'now' );
+		$event_display = Arr::get( $context_arr, 'event_display_mode', Arr::get( $context_arr, 'event_display' ), 'current' );
 
-			$next_event = $next->first();
-			$has_next = $next->found();
+		$args['date_overlaps'] = [ tribe_beginning_of_day( $date ), tribe_end_of_day( $date ) ];
 
-			if ( ! $has_next ) {
-				return '';
-			}
+		return $args;
+	}
 
-			$next_date = tribe_get_start_date( $next_event, false, Dates::DBDATEFORMAT );
-		}
+	/**
+	 * Builds the Day View URL for a specific date.
+	 *
+	 * This is the method underlying the construction of the previous and next URLs.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed $url_date          The date to build the URL for, a \DateTime object, a string or a timestamp.
+	 * @param bool  $canonical         Whether to return the canonical (pretty) version of the URL or not.
+	 * @param array $passthru_vars     An optional array of query variables that should pass thru the method untouched
+	 *                                 in key in value.
+	 *
+	 * @return string The Day View URL for the date.
+	 */
+	protected function build_url_for_date( $url_date, $canonical, array $passthru_vars = [] ) {
+		$url_date        = Dates::build_date_object( $url_date );
+		$url = new Url( $this->get_url() );
```

Comment:
Maybe indent `$url` with the others for better readability.

Reply:
Done, thanks.